contracts/src/v0.8/dev/CanaryUpkeep.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "../interfaces/KeeperCompatibleInterface.sol";
+import "../interfaces/KeeperRegistryInterface.sol";
+import "../ConfirmedOwner.sol";
+
+/**
+ * @notice A canary upkeep which requires a different keeper to service its upkeep at an interval. This makes sure that
+ * all keepers are in a healthy state.
+ */
+contract CanaryUpkeep is KeeperCompatibleInterface, ConfirmedOwner {
+  uint256 private s_keeperIndex;
+  uint256 private s_interval = 300;
+  uint256 private s_timestamp;
+  KeeperRegistryInterface private immutable s_keeperRegistry;
+
+  /**
+   * @param keeperRegistry address of a keeper registry
+   */
+  constructor(KeeperRegistryInterface keeperRegistry) ConfirmedOwner(msg.sender) {
+    s_keeperRegistry = keeperRegistry;
+    s_timestamp = block.timestamp;
+    s_keeperIndex = 0;
+  }
+
+  /**
+   * @return the current keeper index
+   */
+  function getKeeperIndex() external view returns (uint256) {
+    return s_keeperIndex;
+  }
+
+  /**
+   * @return the current timestamp
+   */
+  function getTimestamp() external view returns (uint256) {
+    return s_timestamp;
+  }
+
+  /**
+   * @return the current interval
+   */
+  function getInterval() external view returns (uint256) {
+    return s_interval;
+  }
+
+  /**
+   * @notice updates the interval
+   * @param interval the new interval
+   */
+  function setInterval(uint256 interval) external onlyOwner {
+    s_interval = interval;
+  }
+
+  /**
+   * @notice returns true if keeper array is not empty and sufficient time has passed
+   */
+  function checkUpkeep(
+    bytes calldata /* checkData */
+  ) external view returns (bool, bytes memory) {
+    (State memory _s, Config memory _c, address[] memory keepers) = s_keeperRegistry.getState();
+    bool upkeepNeeded = keepers.length != 0 && block.timestamp >= s_interval + s_timestamp;
```

Comment:
I'm wondering if we really need the condition `keepers.length != 0`

If keepers.length is 0 then there's a problem with the registry and we should surface it

Reply:
yeah i think this makes sense.